### DISCUSSION TITLE
[input] How to use v-money3 with <el-input /> (ElementUI)?

### DISCUSSION THREAD

**Discussion by leeobrum** [element-plus/element-plus#10998] [Feature Request]
Feature type: **`Other`**

## Details

- Enhanced component name: `input`

## Description

Hi, I'm migration my project to vue3 and pinia etc... but don't working v-money3 with my template ElementUI. My old project where I used v-money only, is working. Exemple for error: 
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'length') at B (v-money3.mjs:204:19) 

el-input v-model.lazy="oFormularioCadastro.dbAluguelEmpresa" v-money3="maskMoneyParam"

script setup lang="ts" const maskMoneyParam = reactive({decimal: ',',  thousands: '.',  prefix: 'R$ ',  suffix: '',  precision: 2, masked: false })
Main.ts
import { Money3Directive } from 'v-money3'
const app = createApp(App)
app.directive('money3', Money3Directive)

### Additional comments

_(empty)_

**Comment by diegosilvaNSC** [comment]
Estou com o mesmo problema usando ts e vue3

**Comment by rflcnunes** [comment]
I had the same problem, I managed to solve it using this way:
![image](https://github.com/element-plus/element-plus/assets/58157026/4fb96630-d9f5-4d77-99ed-6f521ae2ad22)

In the script with the config:
![image](https://github.com/element-plus/element-plus/assets/58157026/ae02f182-0e03-4dce-a62d-ce566c106c00)

![image](https://github.com/element-plus/element-plus/assets/58157026/158af7f3-46e5-4776-bc88-04869e9863c3)

Imports:
`import { unformat, format } from 'v-money3'`

**Comment by leeobrum** [comment]
Hi,

I resolved the problem using:

```
<money3 class="sMoney3" inputmode="numeric" maxlength="12" v-model="limiteDiretor" v-bind="maskMoneyParam" />

<script setup lang="ts">
  const maskMoneyParam = reactive({
    decimal: ',',
    thousands: '.',
    prefix: 'R$ ',
    suffix: '',
    precision: 2,
    masked: false,
    disableNegative: true
  })
</script>


.sMoney3 {
  width: 100%;
  height: 37px;
  color: #606266;
  padding: 2px 11px;
  font-size: inherit;
  border: 1px solid #dcdfe6;
  border-radius: 4px;
}
```


Creating an SCSS compatible with el-input and I open one Issue at component v-money3

https://github.com/jonathanpmartins/v-money3/issues/74
